File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/04.For-Loop/03-additional/08.equalPairs.js
function equalPairs(input) {

    const coupleCount = Number(input[0]);
    let index = 1;

    let currentCoupleValue = 0;
    let lastCoupleValue = 0;
    let currentDifference = 0;
    let maxDifference = 0;
    let isHaveDifference = false;

    for (let i = 1; i <= coupleCount; i++) {
        let currentNum1 = Number(input[index++]);
        let currentNum2 = Number(input[index++]);
        currentCoupleValue = currentNum1 + currentNum2;
        if (index <= 3) {
            lastCoupleValue = currentCoupleValue;
        } else if (currentCoupleValue !== lastCoupleValue) {
            isHaveDifference = true;
            currentDifference = Math.abs(currentCoupleValue - lastCoupleValue);
            lastCoupleValue = currentCoupleValue;
            if (currentDifference > maxDifference) {
                maxDifference = currentDifference;
            }
        } else {
            lastCoupleValue = currentCoupleValue;
        }
    }

    if (isHaveDifference) {
        console.log(`No, maxdiff=${maxDifference}`);
    } else {
        console.log(`Yes, value=${lastCoupleValue}`);
    }
}
equalPairs(["2", "-1", "2", "0", "-1"]);

File: Python_Basics/Homeworks-And-Labs/04.for_loop/03.additional/08.equal_pairs.py
couple_count = int(input())

current_couple_value = 0
last_couple_value = 0
current_difference = 0
max_difference = 0
is_have_difference = False
index = 1

for i in range(couple_count):
    current_num_one = int(input())
    current_num_two = int(input())
    index += 2
    current_couple_value = current_num_one + current_num_two
    if index <= 3:
        last_couple_value = current_couple_value
    elif current_couple_value != last_couple_value:
        is_have_difference = True
        current_difference = abs(current_couple_value - last_couple_value)
        last_couple_value = current_couple_value
        if current_difference > max_difference:
            max_difference = current_difference
    else:
        last_couple_value = current_couple_value
    
if is_have_difference:
    print(f'No, maxdiff={max_difference}')
else:
    print(f'Yes, value={last_couple_value}')
